content/tutorials/reactjs-realtime-commenting.textile

```diff
@@ -0,0 +1,344 @@
+
+h2. Step 4 - Create Component Folder
+
+Now, we’ll create a new folder within the @src@  folder called @components@. This folder will contain all our React components.
+
+```[sh]
+cd src
+mkdir components
+```
+
+Once that is created, we need to move @App.js@ into the @components@ folder:
+
+```[sh]
+cd src
+mv App.js components
+```
+
+h2. Step 5 - Update index.js
+
+Then we need to update @src/index.js@ as below:
+
+```[javascript]
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './components/App';
+import registerServiceWorker from './registerServiceWorker';
+
+ReactDOM.render(<App />, document.getElementById('root'));
+registerServiceWorker();
+```
+
+h2. Step 6 - Add Bulma CSS Framework
+
+We’ll be using "Bulma":https://bulma.io Bulma CSS framework so as not to waste time writing CSS. So we’ll reference it from CDN. Add the line below to @public/index.html@:
+
+```[html]
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bulma/0.6.0/css/bulma.min.css">
+```
+
+h2. Step 7 - Install Ably
+
+To start using Ably you first need to install the NPM module. The NPM module can be installed as follows:
+
+```[sh]
+npm install ably
+```
+
+The client library must be instanced with the API key you copied in Step 1. We’ll create a dedicated file for this so we can easily use it in multiple places. Within @src@ folder, create a new file named @ably.js@ and add the code below into it:
+
+```[javascript]
+import { Realtime } from 'ably/browser/static/ably-commonjs.js';
+
+export default new Realtime('INSERT-YOUR-API-KEY-HERE');
+```
+
+h2. Step 8 - Create CommentBox Component
+
+Within the @components@ folder, create a new file named @CommentBox.js@ and add the code below into it:
+
+```[javascript]
+import React, { Component } from 'react';
+import Ably from '../ably';
+
+class CommentBox extends Component {
+  render() {
+    return (
+      <div>
+        <h1 className="title">Kindly leave your thoughts below</h1>
+        <form onSubmit={this.addComment}>
+          <div className="field">
+            <div className="control">
+              <input type="text" className="input" name="name" placeholder="Your name"/>
+            </div>
+          </div>
+          <div className="field">
+            <div className="control">
+              <textarea className="textarea" name="comment" placeholder="Add a comment"></textarea>
+            </div>
+          </div>
+          <div className="field">
+            <div className="control">
+              <button className="button is-primary">Submit</button>
+            </div>
+          </div>
+        </form>
+      </div>
+    );
+  }
+}
+
+export default CommentBox;
+```
+
+This simply renders a comment form as the image below. Once the form is submitted, we trigger an @onSubmit@ event which in turn call the @addComment()@ function (which we’ll create shortly).
```

Comment:
I already mentioned it renders a comment form.

Reply:
Well sort of.  The problem is at this point it does not because its not invoked or reference anywhere.  I don't think this explains to a noob why you're doing this first and how it renders.